python/cudf/cudf/core/dataframe.py

```diff
@@ -1339,8 +1339,17 @@ def _slice(self: T, arg: slice) -> T:
 
     @_cudf_nvtx_annotate
     def memory_usage(self, index=True, deep=False):
-        return Series(
-            {str(k): v for k, v in super().memory_usage(index, deep).items()}
+        return Series._from_data(
+            data={
+                None: as_column(
+                    [col.memory_usage for col in self._data.columns]
+                    + ([self.index.memory_usage()] if index else [])
```

Comment:
FWIW if we're really trying to squeeze the last drop of performance out of this we'll probably get another ~50 ns by accessing `self._index` instead of `self.index`.

Reply:
```suggestion
                    + ([self._index.memory_usage()] if index else [])
```